src/parsing.jl

```diff
@@ -364,6 +357,53 @@ function Base.parse(::Type{Dates.CompoundPeriod}, pqv::PQValue{PQ_SYSTEM_TYPES[:
     return Dates.CompoundPeriod(periods)
 end
 
+## ranges
+_DEFAULT_TYPE_MAP[:int4range] = Interval{Int32}
+_DEFAULT_TYPE_MAP[:int8range] = Interval{Int64}
+_DEFAULT_TYPE_MAP[:tsrange] = Interval{DateTime}
+_DEFAULT_TYPE_MAP[:tstzrange] = Interval{ZonedDateTime}
+_DEFAULT_TYPE_MAP[:daterange] = Interval{Date}
+
+const RANGE_REGEX = r"[^\[\(\]\),]+"
```

Comment:
@iamed2 I changed it, not sure if it's exactly what you were going for but take a look and let me know.

Also `const RANGE_ITEM = "[^\[\(\]\),]+"` doesn't work in julia because of string escaping (can't be a string)

Reply:
Ah yeah you'd have to double-escape the backslashes